packages/compass-query-bar/src/components/query-history/favorite-list.tsx

```diff
@@ -54,11 +60,13 @@ const FavoriteItem = ({
       onClick={onCardClick}
       data-testid="favorite-query-list-item"
       header={(isHovered: boolean) => (
-        <QueryItemHeading title={query._name} isHovered={isHovered}>
+        <QueryItemHeading title={query._name} isHovered={true}>
           <CopyActionButton
+            visible={isHovered}
             onClick={() => copyToClipboard(formatQuery(attributes))}
           />
-          <DeleteActionButton onClick={onDeleteClick} />
+          <DeleteActionButton visible={isHovered} onClick={onDeleteClick} />
+          <OpensInModal visible={isUpdateQuery} />
```

Comment:
While it's nice that onClick does the same thing, I'd suggest we pass the callback here to be explicit about what action this button does instead of relying on card click handler for that

Reply:
Makes sense, I'll propagate the handler downwards.